openapi.go

```diff
@@ -47,35 +58,79 @@ func (s *Server) GenerateOpenAPI() {
 	}
 
 	// Write spec to docs/openapi.json
-	err = os.MkdirAll("docs", 0o750)
-	if err != nil {
-		slog.Error("Error creating docs directory", "error", err)
-	}
-	f, err := os.Create("docs/openapi.json")
-	if err != nil {
-		slog.Error("Error creating docs/openapi.json", "error", err)
+	if options.disableSaveOnDisk == false {
+		jsonLocalPath := "docs/openapi.json"
+		if options.jsonLocalPath != "" {
+			jsonLocalPath = options.jsonLocalPath
+
+			match, _ := regexp.MatchString(`^[\.a-zA-Z0-9\-\_]+(.json)$`, jsonLocalPath)
+
+			if match == false {
+				err := errors.New("Value of 'jsonLocalPath' option is not valid")
+				slog.Error("Error writing json spec", "error", err)
+				return
+			}
+		}
+
+		jsonFolder := filepath.Dir(jsonLocalPath)
+		err = os.MkdirAll(jsonFolder, 0o750)
+		if err != nil {
+			slog.Error("Error creating docs directory", "error", err)
+		}
+		f, err := os.Create(jsonLocalPath)
+		if err != nil {
+			slog.Error("Error creating "+jsonLocalPath, "error", err)
+		}
+		defer f.Close()
+		_, err = f.Write(dataJSON)
+		if err != nil {
+			slog.Error("Error writing file", "error", err)
+		}
+		slog.Info("Updated " + jsonLocalPath)
 	}
-	defer f.Close()
-	_, err = f.Write(dataJSON)
-	if err != nil {
-		slog.Error("Error writing file", "error", err)
+
+	if options.disableSwagger == false {
+		// Serve spec as JSON
+		jsonUrl := "/swagger/doc.json"
+		if options.jsonUrl != "" {
+			jsonUrl = options.jsonUrl
+
+			match, _ := regexp.MatchString(`^\/[\/\.a-zA-Z0-9\-\_]+(.json)$`, jsonUrl)
+
+			if match == false {
+				err := errors.New("Value of 'jsonUrl' option is not valid")
+				slog.Error("Error serving openapi json spec", "error", err)
+				return
+			}
+		}
+
+		GetStd(s, jsonUrl, func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+			_, _ = w.Write(dataJSON)
+		})
+
+		// Swagger UI
+		swaggerUrl := "/swagger"
+		if options.swaggerUrl != "" {
+			swaggerUrl = options.swaggerUrl
+
+			match, _ := regexp.MatchString(`^\/[\/\.a-zA-Z0-9\-\_]+[\.a-zA-Z0-9\-\_]$`, swaggerUrl)
```

Comment:
It's nice to have a param validation ! Very good idea 🎉 
1. Note that building a regex takes time: don't do it in a loop, for example. In this case, you only have to do it once, so don't worry. Alternatively, you can compile it at build time, as a local variable, as in the `params.go` file, instead of building it at runtime.
2. Every regex must have test cases. Sorry but we're making a premium quality framework! 😉